string_grouper/test/test_string_grouper.py

```diff
@@ -124,118 +155,138 @@ def test_case_insensitive_build_matches_list(self):
     def test_get_matches_two_dataframes(self):
         test_series_1 = pd.Series(['foo', 'bar', 'baz'])
         test_series_2 = pd.Series(['foo', 'bar', 'bop'])
-        sg = StringGrouper(test_series_1, test_series_2).fit()
+        # sg = StringGrouper(test_series_1, test_series_2).fit()
```

Comment:
Sorry - I like the old way better :) 

These tests were testing the functionality of the StringGrouper.get_matches function, not the match_strings function. 

The match_strings function happens to call this function, which is why you reach 100% code coverage, however now we are not testing the get_matches function anymore. 

To get to 100% we need to test what the match_strings function does: create a StringGrouper object and call the get_matches function (which has already been tested so no need to test it again). For this you would need to use something like MagicMock. I have created an example for this once here: https://gist.github.com/Bergvca/069d50d58b56fba0f9ee695d92ccebde

Reply:
I must confess I was somewhat lazy with the unit tests.  Will apply your suggestions.  
Good to know about MagicMock!  I'll take a closer look.

Reply:
I've used unittest.mock to mock the StringGrouper class and some of its methods in order to simulate the high-level functions and achieve 100% code coverage.  

First time hearing of 'mocking'.  Interesting stuff! Thanks for bringing that to my attention.

Reply:
Yes mocking can be very useful, especially when you are testing a function that requires a complicated object, like a database object.